backend/tests/integration-async/assign/test_views.py

```diff
@@ -0,0 +1,77 @@
+import orjson
+from helpers.providers.faker import faker
+from helpers.utils import create_user
+from invenio_accounts.testutils import login_user_via_session
+from invenio_db import db
+from sqlalchemy.exc import ResourceClosedError
+
+from inspirehep.records.api import AuthorsRecord, LiteratureRecord
+
+
+def test_assign_doesnt_craise_resources_closed_error(inspire_app, override_config):
+    with override_config(
+        FEATURE_FLAG_ENABLE_BAI_PROVIDER=True, FEATURE_FLAG_ENABLE_BAI_CREATION=True
+    ):
+        author_record_data = faker.record("aut")
+        author_record = AuthorsRecord.create(author_record_data)
+
+        paper_data = faker.record("lit")
+        paper_data.update(
+            {
+                "authors": [
+                    {
+                        "full_name": author_record["name"]["value"],
+                        "record": author_record["self"],
+                    }
+                ]
+            }
+        )
+        paper = LiteratureRecord.create(paper_data)
+
+        paper_data_2 = faker.record("lit")
+        paper_data_2.update(
+            {
+                "authors": [
+                    {
+                        "full_name": author_record["name"]["value"],
+                        "record": author_record["self"],
+                    }
+                ]
+            }
+        )
+        paper_2 = LiteratureRecord.create(paper_data_2)
+
+        paper_data_3 = faker.record("lit")
+        paper_data_3.update(
+            {
+                "authors": [
+                    {
+                        "full_name": author_record["name"]["value"],
+                        "record": author_record["self"],
+                    }
+                ]
+            }
+        )
+        paper_3 = LiteratureRecord.create(paper_data_3)
+        db.session.commit()
+
+        cataloger = create_user(role="cataloger")
+        try:
+            with inspire_app.test_client() as client:
+                login_user_via_session(client, email=cataloger.email)
+                client.post(
+                    "/assign/author",
+                    data=orjson.dumps(
+                        {
+                            "literature_recids": [
+                                paper["control_number"],
+                                paper_2["control_number"],
+                                paper_3["control_number"],
+                            ],
+                            "from_author_recid": author_record["control_number"],
+                        }
+                    ),
+                    content_type="application/json",
+                )
+        except ResourceClosedError:
+            assert False
```

Comment:
are you sure this test is failing without your change? I tried something extremely similar yesterday on a worker and I didn't get any exception.